KalturaWowzaServer/src/main/java/com/kaltura/media/server/wowza/listeners/LiveStreamEntry.java

```diff
@@ -1017,4 +1054,70 @@ public void onHTTPSessionCreate(IHTTPStreamerSession httpStreamerSession){
 		}
 
 	}
+
+    class ChunkIdHandler implements IHTTPStreamerCupertinoLiveStreamPacketizerChunkIdHandler
+    {
+        private LiveStreamPacketizerCupertino packetizerCupertino;
+        // TO DO: Think how to open/re-open that flag after onUnPublish zv
+        private boolean eventLunched = false;
+
+        public ChunkIdHandler(LiveStreamPacketizerCupertino packetizer) {
+            this.packetizerCupertino = packetizer;
+        }
+
+        public void init(LiveStreamPacketizerCupertino var) {}
+
+        public long onAssignChunkId(HTTPStreamerCupertinoLiveStreamPacketizerChunkIdContext dummyVar) {
+            if (!eventLunched) {
+                checkChunksForPlayback();
+            }
+            return dummyVar.getChunkIndex();
+        }
+
+        public void checkChunksForPlayback() {
+            try {
+                int chunkCount = this.packetizerCupertino.getChunkCount();
+                logger.info("Stream contains [" + chunkCount + "] chuncks");
+                if (chunkCount >= readyForPlaybackMinimumChunkCount) {
+                    logger.info("Raising STREAM_READY_FOR_PLAYBACK event");
+                    raiseReadyForPlaybackEvent();
+                }
+            } catch (Exception err) {
+                logger.error(err);
+            }
+        }
+
+        private void raiseReadyForPlaybackEvent() {
+            IMediaStream mediaStream = this.packetizerCupertino.getAndSetStartStream(null);
```

Comment:
Not sure we need to. If the packetizerCupertino is null we will never create an instance of ChunkIdHandler (check the LiveStreamPacketizerListener2:onLiveStreamPacketizerCreate method where we check if the packetizer is instance of LiveStreamPacketizerCupertino)